src/App.tsx

```diff
@@ -15,48 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+interface SortFields {
+  sortField: SortType,
+  isReverse: boolean,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReverse }: SortFields,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField !== SortType.Default) {
+    switch (sortField) {
+      case SortType.Alphabet:
+        return preparedGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+      case SortType.Length:
+        return preparedGoods
+          .sort((goodA, goodB) => goodA.length - goodB.length);
+      default:
+        return preparedGoods;
+    }
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.Default);
+  const [isReverse, setIsReverse] = useState<boolean>(false);
+  const visibleGoods: string[] = getPreparedGoods(goodsFromServer,
+    { sortField, isReverse });
+
+  const handlerOnReset = () => {
+    setSortField(SortType.Default);
+    setIsReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.Alphabet,
+          })}
```

Comment:
```suggestion
          className={cn(
            'button',
            'is-info',
            { 'is-light': sortField !== SortType.Alphabet },
          )}
```
Let's improve the code style a bit, change in all places